src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -577,6 +580,39 @@ public boolean hasTag(String tag) {
 				log.printError();
 				return null;
 			}
+			Matcher literalClassInfoMatcher = LITERAL_CLASSINFO_PARSE.matcher(expr);
+			if (literalClassInfoMatcher.matches()) {
+				String literalString = literalClassInfoMatcher.group(1);
+				String classInfoString = literalClassInfoMatcher.group(2);
+				ClassInfo<?> classInfo = Classes.parse(classInfoString, ClassInfo.class, context);
+				if (classInfo == null) {
+					log.printError();
+					return null;
+				}
+				Parser<?> classInfoParser = classInfo.getParser();
+				if (classInfoParser == null || !classInfoParser.canParse(context)) {
+					Skript.error("A '" + classInfoString  + "' cannot be parsed.");
```

Comment:
I'm thinking it would actually be better to not error here and continue parsing instead. What if there is a registered syntax that happens to match this same pattern? I admit I'm not entirely sure if this would cause Skript not check other syntax but I'm fairly sure it would

Reply:
After this is only other literals and a variable string check left, so I think it's fine to exit here. All other expressions have already been tried.

Reply:
i'm still a little uneasy about it but i'll defer to you on this one

Reply:
![image](https://github.com/user-attachments/assets/f27e4263-f101-44ee-af26-9af6ee82f2cc)